Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -2488,6 +2488,37 @@ def visit_CClassDefNode(self, node):
         return node
 
 
+class AutoCpdefFunctionDefinitions(CythonTransform):
+
+    def visit_ModuleNode(self, node):
+        self.scope = node.scope
+        self.directives = node.directives
+        self.imported_names = set()  # hack, see visit_FromImportStatNode()
+        self.visitchildren(node)
+        return node
+
+    def visit_DefNode(self, node):
+        pxd_def = self.scope.lookup(node.name)
+        if pxd_def and (not pxd_def.scope or not pxd_def.scope.is_builtin_scope):
+            pass
```

Comment:
This is quick and ugly. I left it for most straightforward change.

Reply:
Just note: The PR currently is not 100% cleaned. I would like to understand whether the approach is fine.

Reply:
The only issue with the approach is that each transform takes time to run, so we're trying to avoid too many transforms for small tasks (although I'm a bit guilty of doing that too).

This particular transform looks like it only does a fairly shallow run (it doesn't actually go into functions for example) so it may not be too bad. It should duplicate `visit_ExprNode` from `AlignFunctionDefinitions` though (because it never cares about anything in an expression).

I think it's probably OK (mainly because it can't obviously be merged with any of the neighbouring transforms)...